### PR TITLE
feat: Added grouped series and documented how to render stacked series

### PR DESCRIPTION
**Pull request by ColinEberhardt**
This adds a grouped series implementation (much the same as the one in the original d3fc, just tidied up a bit) for SVG and canvas.

For stacked the original d3fc is somewhat trivial, so rather than replicate the component, I've instead turned it into an example.

The example code depends on [d3fc-spread](https://github.com/ColinEberhardt/d3fc-spread).

After looking at the changes in the d3 stack component, I think the d3fc-spread API is still correct and of value.

### CODE REVIEW

**Comment by ColinEberhardt** [comment]
NOTE: Build fails because d3fc-spread has not been released yet

**Comment by ColinEberhardt** [comment]
d3fc-group has now been released, so this PR passes the build

File: src/svg/grouped.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export default (series) => {
+
+    const base = groupedBase(series);
+
+    const join = dataJoin('g', 'grouped');
+
+    const grouped = (selection) => {
+        selection.each((data, index, group) => {
+            const groupWidth = base.computeGroupWidth(data);
+            const offsetScale = scaleLinear();
```
Comment (chrisprice) [inline src/svg/grouped.js]: Would an ordinal scale allow more flexibility here? I think it also contains a lot of the offset logic.

File: src/svg/grouped.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+export default (series) => {
+
+    const base = groupedBase(series);
+
+    const join = dataJoin('g', 'grouped');
+
+    const grouped = (selection) => {
+        selection.each((data, index, group) => {
+            const groupWidth = base.computeGroupWidth(data);
```
Comment (chrisprice) [inline src/svg/grouped.js]: Can this not be moved into the base?

Comment (chrisprice) [inline src/svg/grouped.js]: Down to the end of the offsetScale configuration

**Comment by ColinEberhardt** [comment]
@chrisprice - updated as per discussion

File: src/groupedBase.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    let xScale = scaleLinear();
+
+    const offsetScale = scaleBand();
+    const grouped = () => {};
+
+    const computeGroupWidth = (data) => {
+        if (!data.length) {
+            return 0;
+        }
+        const seriesData = data[0];
```
Comment (chrisprice) [inline src/groupedBase.js]: Question - is it ok that we throw away the rest of the data when calculating the width?

Comment (ColinEberhardt) [inline src/groupedBase.js]: see #27

**Pull request merged**